src/SeleniumLibrary/keywords/element.py

```diff
@@ -693,25 +715,31 @@ def page_should_not_contain_image(self, locator, message=None, loglevel='INFO'):
 
     @keyword
     def get_matching_xpath_count(self, xpath, return_str=True):
-        """Returns number of elements matching ``xpath``.
-
-        The default return type is string, but it can changed to an integer
-        by setting the ``return_str`` argument to a false value.
-
-        The ``xpath`` should not contain ``xpath:`` prefix.
-
-        Example:
-        | count = | `Get Matching Xpath Count` | //div[@id='sales-pop'] |
-        """
-        count = len(self.find_elements('xpath:' + xpath))
+        """Deprecated. Use `Get Element Count` instead."""
+        count = self.get_element_count('xpath:' + xpath)
         return str(count) if is_truthy(return_str) else count
 
     @keyword
     def xpath_should_match_x_times(self, xpath, x, message=None, loglevel='INFO'):
-        """Deprecated. Use `Locator Should Match X Times` instead."""
+        """Deprecated, use `Page Should Contain Element` with `limit` argument instead."""
         self.locator_should_match_x_times('xpath:'+xpath, x, message, loglevel)
 
     @keyword
+    def get_element_count(self, locator):
+        """Returns number of elements matching ``locator``.
+
+        If you wish to assert the number of matching elements, use
+        `Page Should Contain Element` with `limit` argument. Keyword will
+        always return an integer.
+        New in SeleniumLibrary 3.0.
```

Comment:
Also this `New in ...` note would, in my opinion, work better at the very end of the kw doc after examples. Also Python has them there as this example illustrates: https://docs.python.org/3/library/argparse.html#allow-abbrev